session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSessionRegistry.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cassandra.session.scaladsl
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.cassandra.session.CqlSessionProvider
+import akka.event.Logging
+import com.datastax.oss.driver.api.core.CqlSession
+
+/**
+ * This Cassandra session registry makes it possible to share Cassandra sessions between multiple use sites
+ * in the same `ActorSystem` (important for the Cassandra Akka Persistence plugin where it is shared between journal,
+ * query plugin and snapshot plugin)
+ */
+object CassandraSessionRegistry extends ExtensionId[CassandraSessionRegistry] with ExtensionIdProvider {
+
+  def createExtension(system: ExtendedActorSystem): CassandraSessionRegistry =
+    new CassandraSessionRegistry(system)
+
+  /**
+   * Java API: get the session registry
+   */
+  override def get(system: ActorSystem): CassandraSessionRegistry =
+    super.get(system)
+
+  override def lookup(): ExtensionId[CassandraSessionRegistry] = this
+
+  private case class SessionKey(configPath: String)
```

Comment:
do we need add the other properties to the key? Otherwise two calls with different ECs will end up with a single session using one of the ECs

Reply:
I was thinking about the ec but didn't consider it important enough for including in the key. I was thinking that the ec of the CassandraSession doesn't matter much. It's only for the `map`/`flatMap` operations there. Everything heavy will be later in the streams/actors and those use `plugin-dispatcher`.

Reply:
:+1: